jmix-flowui/flowui/src/main/java/io/jmix/flowui/app/propertyfilter/dateinterval/DateIntervalDialog.java

```diff
@@ -0,0 +1,272 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.flowui.app.propertyfilter.dateinterval;
+
+import com.vaadin.flow.component.AbstractField.ComponentValueChangeEvent;
+import com.vaadin.flow.component.ClickEvent;
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
+import io.jmix.core.Messages;
+import io.jmix.core.metamodel.model.MetaPropertyPath;
+import io.jmix.core.metamodel.model.Range;
+import io.jmix.flowui.UiComponents;
+import io.jmix.flowui.app.propertyfilter.dateinterval.model.BaseDateInterval;
+import io.jmix.flowui.app.propertyfilter.dateinterval.component.LastIntervalField;
+import io.jmix.flowui.app.propertyfilter.dateinterval.component.NextIntervalField;
+import io.jmix.flowui.app.propertyfilter.dateinterval.model.BaseDateInterval.Type;
+import io.jmix.flowui.app.propertyfilter.dateinterval.model.DateInterval;
+import io.jmix.flowui.app.propertyfilter.dateinterval.model.DateInterval.TimeUnit;
+import io.jmix.flowui.app.propertyfilter.dateinterval.model.predefined.PredefinedDateInterval;
+import io.jmix.flowui.app.propertyfilter.dateinterval.model.predefined.PredefinedDateIntervalRegistry;
+import io.jmix.flowui.component.checkboxgroup.JmixCheckboxGroup;
+import io.jmix.flowui.component.radiobuttongroup.JmixRadioButtonGroup;
+import io.jmix.flowui.component.select.JmixSelect;
+import io.jmix.flowui.component.validation.ValidationErrors;
+import io.jmix.flowui.kit.component.ComponentUtils;
+import io.jmix.flowui.kit.component.button.JmixButton;
+import io.jmix.flowui.view.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.lang.Nullable;
+
+import java.sql.Time;
+import java.time.LocalTime;
+import java.time.OffsetTime;
+import java.util.*;
+
+@ViewController("flowui_DateIntervalDialog")
+@ViewDescriptor("date-interval-dialog.xml")
+@DialogMode(height = "AUTO", width = "50em", resizable = true)
```

Comment:
1. isn't height AUTO by default?
2. As I remember, we decided to make it resizable=false

Reply:
3. 
<img width="850" alt="Screenshot 2024-04-01 at 11 41 21" src="https://github.com/jmix-framework/jmix/assets/6077149/436d4b8f-d9ad-4e31-aeed-4dde88c5ed6a">

Reply:
1. Yep, u are right(
2. Yes, cause when the dialog is displayed on a mobile device, it takes on the maximum available width by default. Fixed.
3. I do vanga, u haven't installed `jmix-data` module, which contains the `RelativeDateTimeMoment` enumeration and its localization

Reply:
but I can select *relative date and time* without this module. Why do I need to install additional module just to get a localized messages?

Reply:
No, you cannot select the _relative date and time_ without this module. The entity that will be processed on the backend is determined in this module
`RelativeDateTimeMoment` location - `io.jmix.data.impl.queryconstant`
Therefore the localization is in the `jmix-data` module